### DISCUSSION TITLE
Generate sideset and nodeset for the intersection of the subdomains

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#20310] [Q&A Modules: General] [answered]
Hello MOOSE Experts

I am importing the Abaqus mesh file in the MOOSE. There is a number of subdomains in it. 

Is it possible to generate the NODESETS for the nodes that are at the intersection between subdomains? Also, I want to generate the SIDESET of the outer boundary of the entire domain. 

Please guide me through it. I shall be very thankful.

Best
Avtar

**Comment by GiudGiud** [comment, marked as answer]
Hello

See the mesh generators for generating sidesets, notably these two:
https://mooseframework.inl.gov/source/meshgenerators/SideSetsBetweenSubdomainsGenerator.html
https://mooseframework.inl.gov/source/meshgenerators/SideSetsAroundSubdomainGenerator.html
And see the options in this page to generate nodesets from sidesets:
https://mooseframework.inl.gov/source/mesh/MooseMesh.html

Guillaume

**Reply by GiudGiud** [reply]
What is the domain outer boundary? If continuous, might be easier to just use that one for the outer boundary
https://mooseframework.inl.gov/source/meshgenerators/SideSetsFromNormalsGenerator.html

**Reply by avtarsinghh1991** [reply]
Hello @GiudGiud 

It is multiple 3d grains intersecting each other.

**Reply by avtarsinghh1991** [reply]
Hello @GiudGiud 

I am trying the following
https://mooseframework.inl.gov/source/meshgenerators/SideSetsBetweenSubdomainsGenerator.html

But I do not have the information that the given subdomain is surrounded by how many and which subdomains.
I think in this way, I cannot able to find all the intersection Nodesets once. For this, I need to apply it for individual subdomains. Correct me if I am wrong.

Best
Avtar

**Reply by GiudGiud** [reply]
You can list one primary and **all** the other blocks in paired_block to be sure (you ll have to list them), even if they dont touch
Or you can use https://mooseframework.inl.gov/source/meshgenerators/SideSetsAroundSubdomainGenerator.html
if you just want to have what s around one grain.
